File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    for i in range(len(arr1)):
        answer.append(format(arr1[i]|arr2[i],'b').zfill(n))
        answer[i]=answer[i].replace("1","#")
        answer[i]=answer[i].replace("0"," ")
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for (int i = 0; i < n; i++){
            String s1 = String.format("%16s", Integer.toBinaryString(arr1[i])).replace(" ","0").substring(16 - n);
            String s2 = String.format("%16s", Integer.toBinaryString(arr2[i])).replace(" ","0").substring(16-n);
            StringBuilder sb = new StringBuilder();
            
            for (int j = 0; j < n; j++){
                if (s1.charAt(j) == '1' || s2.charAt(j) == '1') sb.append("#");
                else sb.append(" ");
            }
            answer[i] = sb.toString();
            
        }
        return answer;
    }
}